cupyx/scipy/sparse/_data.py

```diff
@@ -123,13 +123,13 @@ def power(self, n, dtype=None):
 def _find_missing_index(ind, n):
     for k, a in enumerate(ind):
         if k != a:
```

Comment:
It still looks that device-to-host sync is caused here.
How about comparing `ind` and `cupy.arange(ind.size)` and finding the minimum index `i` such that `ind[i] != i`, without synchronization?